pkg/admission/init.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package admission
+
+import (
+	"reflect"
+	"k8s.io/kubernetes/pkg/controller/framework"
+	"k8s.io/kubernetes/pkg/api"
+)
+
+type PluginInitializer interface {
+	Initialize(plugins []Interface)
```

Comment:
I like this approach. Do we want only one plugin initializer for all plugins? Or do we want more plugin initializers each one for different object (e.g. informer, authorizer, client) being injected?

As there are more informers, I can image:

```go
// Initializer for plugins with shared informer
func initPluginInformers(client clientset.Interface) map[reflect.Type]framework.SharedIndexInformer {
	informers := make(map[reflect.Type]framework.SharedIndexInformer)

	informers[reflect.TypeOf(&api.Namespace{})] = informers.CreateSharedNamespaceIndexInformer(client, 5*time.Minute)
	informers[reflect.TypeOf(&api.Node{})] = informers.CreateSharedNodeIndexInformer(client, 5*time.Minute)
	informers[reflect.TypeOf(&api.Pod{})] = informers.CreateSharedPodIndexInformer(client, 5*time.Minute)
	...

	return informers
}

type pluginInformerInitializer struct {
	informers map [reflect.Type] framework.SharedIndexInformer
}

func NewPluginInformerInitializer(client clientset.Interface) pluginInformerInitializer {
	initializer := &pluginInformerInitializer{
		informers : initPluginInformers(client),
	}

	return initializer
}
```

and more clients:

```go
// Initializer for plugins with clients
func initPluginClients(???) map[???]clientset.Interface {
	clients := make(map[???]clientset.Interface)

	clients[???] = ???.CreateEtcdClient(???)
	clients[???] = ???.CreateKubeClient(???)
	...

	return clients
}

type pluginClientInitializer struct {
	clients map [???]clientset.Interface
}

func NewPluginClientInitializer(???) pluginClientInitializer {
	initializer := &pluginClientInitializer{
		clients : initPluginClients(???),
	}

	return initializer
}
```

Or everything mixed together, then the interface does not make sense:

```go
type pluginInitializer struct {
	informers map [reflect.Type] framework.SharedIndexInformer
	clients map [???]clientset.Interface
}

func NewPluginInitializer(client clientset.Interface) pluginInformerInitializer {
	initializer := &pluginInitializer{
		informers: initPluginInformers(client),
		clients : initPluginClients(???),
	}

	return initializer
}
```